MekHQ/src/mekhq/campaign/GamePreset.java

```diff
@@ -228,53 +234,41 @@ else if (xn.equalsIgnoreCase("campaignOptions")) {
                             continue;
                         }
                         if (!wn2.getNodeName().equalsIgnoreCase("ability")) {
-                            // Error condition of sorts!
-                            // Errr, what should we do here?
-                            MekHQ.getLogger().log(GamePreset.class, METHOD_NAME, LogLevel.ERROR,
-                                    "Unknown node type not loaded in Special Ability nodes: " //$NON-NLS-1$
-                                             + wn2.getNodeName());
+                            MekHQ.getLogger().error(GamePreset.class, METHOD_NAME,
+                                    "Unknown node type not loaded in Special Ability nodes: " + wn2.getNodeName());
                             continue;
                         }
 
-                        SpecialAbility.generateSeparateInstanceFromXML(wn2, preset.specialAbilities, options);
+                        SpecialAbility.generateSeparateInstanceFromXML(wn2, preset.getSpecialAbilities(), options);
                     }
                 }
             }
         }
-		return preset;
-	}
-
-	/**
-	 * Collect and load all the Game Presets in files in a given directory and
-	 * return an ArrayList of them
-	 * @param directory
-	 * @return
-	 */
-	public static ArrayList<GamePreset> getGamePresetsIn(String directory) {
-
-		ArrayList<GamePreset> presets = new ArrayList<GamePreset>();
-
-		File[] files = Utilities.getAllFiles(MekHQ.PRESET_DIR, new FilenameFilter() {
-            public boolean accept(File dir, String name) {
-                return name.toLowerCase().endsWith(".xml");
-            }
-        });
-    	for(File file : files) {
-    		// And then load the campaign object from it.
-    		FileInputStream fis = null;
-    		GamePreset preset = null;
-
-    		try {
-    			fis = new FileInputStream(file);
-    			preset = GamePreset.createGamePresetFromXMLFileInputStream(fis);
-    			if(preset.isValid()) {
-    				presets.add(preset);
-    			}
-    			fis.close();
-    		} catch (Exception e) {
+        return preset;
+    }
+
+    /**
+     * Collect and load all the Game Presets in files in a given directory and
+     * return a List of them
+     * @return a list of all of the game presets in the given directory
+     */
+    public static List<GamePreset> getGamePresetsIn() {
+        List<GamePreset> presets = new ArrayList<>();
+
+        File[] files = Utilities.getAllFiles(MekHQ.PRESET_DIR, (dir, name) -> name.toLowerCase().endsWith(".xml"));
+        for (File file : files) {
+            // And then load the campaign object from it.
+            GamePreset preset;
+
+            try (FileInputStream fis = new FileInputStream(file)) {
```

Comment:
I assume this construct automatically closes the input stream regardless of whether the try/catch block throws an exception or not?

Reply:
Exactly, so you don't have to worry about adding it yourself or not having it closed following an exception.